docs/how_to_guides/hosting_validator_models.md

```diff
@@ -0,0 +1,210 @@
+## Step 1: Pick a Validator
+
+First, choose a validator that you want to use. You can browse a list of all the Guardrails validators on [Guardrails Hub](https://hub.guardrailsai.com/).
+
+:::note
+We suggest picking the “ML” filter on the left side of the page in the “Infrastructure Requirements” section to find validators that use models.
+:::
+
+![alt_text](./assets/infrastructure_requirements.png "image_tooltip")
+
+For the purpose of this guide, let’s use the [ToxicLanguage validator](https://hub.guardrailsai.com/validator/guardrails/toxic_language).
+
+## Step 2: Find and Download the Validator Model
+
+The validator model is referenced in the validator implementation, as well as in the README.
+
+![alt_text](./assets/validator_in_readme.png "image_tooltip")
+
+As we can see, the ToxicLanguage validator currently uses the [Detoxify model](https://github.com/unitaryai/detoxify). In order to use this, we need to install the necessary libraries:
+
+```bash
+# Install FastAPI server dependencies
+pip install fastapi uvicorn
+
+# Option 1 - Install required validator dependencies(General approach - Requires Guardrails)
+guardrails hub install hub://guardrails/toxic_language --install-local-models
+
+# Option 2 - Install only required dependencies (Validator Specific - From README)
+pip install detoxify torch
+```
+
+## Step 3: Wrap the model with FastAPI and run the server
+
+Before implementing the API, we first need to define what the API expects as input and what it will return as output so that the API and server can effectively communicate with each other. 
+
+### The Guardrails Standard Inference Endpoint Schema
+
+Guardrails AI recommends using a similar schema for inputs and outputs across all APIs fronting machine learning models. This approach standardizes the implementation for extracting model inference results out of different models in validators. 
+
+The standard inference request can be found [here](https://github.com/guardrails-ai/guardrails/blob/main/guardrails/validator_base.py#L258). 
+
+The standard input definition looks like this: 
+
+```json
+{
+  "inputs": [
+    {
+      "name": "text",
+      "shape": [1],
+      "data": ["Text we want to check for toxic language"],
+      "datatype": "BYTES"
+    },
+    {
+      "name": "threshold",
+      "shape": [1],
+      "data": [0.5],
+      "datatype": "FP32"
+    }
+  ]
+}
+```
+
+Here, `“text”` contains the text we want to check for toxic language and `“threshold”` contains the confidence threshold for determining toxicity. If the model predicts a toxic level higher than the threshold, then the text is considered toxic. 
+
+The standard output definition looks like this:
+
+```json
+{
+  "model_name": "model_name",
+  "model_version": "1",
+  "outputs": [
+    {
+      "name": "result",
+      "datatype": "BYTES",
+      "shape": [2],
+      "data": ["toxicity", "threat"]
+    }
+  ]
+}
+```
+
+Now it’s time to write our `app.py` to set up the FastAPI server:
+
+```python
```

Comment:
Instead of writing this, we should update the app.py to be a standard flask app and tell people 
1. where to find it
2. how to download it
3. how to run it.
4. How to do this for validators that we don't have an app.py for, and how to identify that

That way, we get a lot of heavy code out of this doc